src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((result, item) => {
+    const keyValue = item[key] as string;
```

Comment:
Type assertion using `as string` is potentially unsafe if `T[keyof T]` is not guaranteed to be a string. You should either ensure by type constraints that the key always results in a string value or handle cases where the value is not a string appropriately.